extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/QuarkusTransactionImpl.java

```diff
@@ -218,6 +218,11 @@ private static void begin(RunOptionsBase options) {
         }
     }
 
+    static void begin(BeginOptions options) {
+        RequestScopedTransaction tx = Arc.container().instance(RequestScopedTransaction.class).get();
```

Comment:
Do you happen to know why the `RequestScopedTransaction` is used instead of `cachedUserTransaction` for the `begin()` method?

Reply:
I guess it's just a failsafe in `QuarkusTransaction#begin`?

It's documented that way, at least:

https://github.com/quarkusio/quarkus/blob/eed0241eaf6dfdd1f148e14dd153edf7404ac2d1/extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/QuarkusTransaction.java#L37-L38

That's not really related to our problem, though. `RequestScopedTransaction` uses a CDI-provided `UserTransaction` under the hood, so it should work fine.